File: Algorithms/SortingSeaching/ShellSort.py

def shell_sort(nums):

    gap = len(nums) // 2

    while gap > 0:

        for i in range(gap, len(nums)):

            j, temp = i, nums[i]

            while j >= gap and nums[j - gap] > temp:

                nums[j] = nums[j - gap]

                j -= gap

            nums[j] = temp

        gap //= 2

def main():

    nums = [1, 52, 2, 25, 6, 10, 25, 200, 15, 12, 8, 2]

    print(nums)
    shell_sort(nums)
    print(nums)

if __name__ == '__main__':
    main()

File: Algorithms/SortingSeaching/ShellSort.js

let shellSort = (nums) => {

    for (let gap = Math.floor(nums.length / 2); gap > 0; gap = Math.floor(gap / 2)) {

        for (let i = gap; i < nums.length; i++) {

            let j = i,
                temp = nums[i];

            while (j >= gap && nums[j - gap] > temp) {

                nums[j] = nums[j - gap];

                j -= gap;

            }

            nums[j] = temp;

        }

    }

}

let nums = [1, 52, 2, 25, 6, 10, 25, 200, 15, 12, 8, 2];

console.log(nums);

shellSort(nums);

console.log(nums);
